stdlib/collections/__init__.pyi

```diff
@@ -266,6 +270,10 @@ class Counter(dict[_T, int], Generic[_T]):
     def __ior__(self: Self, other: Counter[_T]) -> Self: ...  # type: ignore[override]
     if sys.version_info >= (3, 10):
         def total(self) -> int: ...
+        def __le__(self, other: Counter[object]) -> bool: ...
```

Comment:
I think we should use Counter[Any] here. It doesn't work with *all* Counters, only with Counters of a type that's comparable to our type. That's something we can't express in the current type system, so we should use Any.

Reply:
```python
>>> from collections import Counter
>>> Counter('abcde') < Counter((1, 2, 3))
False
```

What am I missing? :)

Reply:
Ah right, the type parameter is the keys, but it compares the values, which are always ints.

This does make for some fun behavior:
```
In [16]: Counter((1, 2, 3)) < Counter('abcde')
Out[16]: False

In [17]: Counter('abcde') < Counter((1, 2, 3))
Out[17]: False
```